bugRzilla_review.Rmd

```diff
@@ -168,45 +215,659 @@ Also there are columns which are empty so they are of no use of the analysis lik
+<font>
+
+```{r Fixed month graph} 
+# plotting graph with creation year where resolution is Fixed
+fixed_month_graph <- ggplot(res_fixed) +
+    geom_bar(aes(x = lastdiffed_month)) +
+    labs(
+        title = "Month in which fixed bugs are last modified",
+        x = "Month ",
+        y = "Bug_Count"
+    )
+ggplotly(fixed_month_graph)
+```
+<font size = 4>
+    The `bar graph` is about the month in which is bugs are last modified where the resolution is `Fixed`. From the graph, we can see that the most wast last modified in the month `December` having a bug count of `559` and in the month of `September` having a bug count of `228` are least modified. This graph is from the year `1998` to `2021`.
+<font>
+
+```{r Invalid bugs graph}
+res_invalid <- baa %>%
+    filter(resolution == "INVALID" & bug_status == "CLOSED")
+
+invalid_graph <- ggplot(res_invalid) +
+    geom_bar(aes(x = creation_month)) +
+    labs(
+        title = "Month in which invalid bugs are filed",
+        x = "Month ",
+        y = "Bug_Count"
+    )
+ggplotly(invalid_graph)
+```
+
+<font size = 4>
+    The `bar graph` is about the month in which is bugs are last modified where `resolution` is `Invalid` and `status` is `closed`. From the graph, we can see that the most were created in the month `October` having a bug count of `625` and in the month of `February` having a bug count of `491` are created. This graph is from the year `1998` to `2021`.
+<font>
+
+```{r Invaild year graph}
+invaild_year_graph <- ggplot(res_invalid) +
+    geom_bar(aes(x = creation_year)) +
+    labs(
+        title = "Year in which INVALID Bugs are Filed",
+        x = "Year",
+        y = "Bug_Count"
+    ) + coord_flip()
+ggplotly(invaild_year_graph)
+```
+
+<font size = 4>
+    The `bar graph` refers to the Creation of Invalid bugs. In the year, `1998` the total of `63` Invalid bugs are created which are least, and in the year `2013` a total of `431` bugs are filed which are most.
+<font>
+
+```{r Priority graph}
+priority_graph <- baa %>%
+    ggplot(aes(x = creation_year, y = bug_id)) +
+    geom_point() +
+    facet_wrap( ~priority) +
+    labs(title = "Bugs created year with their priorities",
+         y = "Bug ID",
+         x = "Date") + theme_bw(base_size = 9) +
+    coord_flip()
+ggplotly(priority_graph)
+```
+
+<font size = 4>
+    The `Lattice plot` gives insight about the bugs when they are created and under which priority they fall like from the above plot we can conclude that the majority of the bugs are filed under the `P5` which is having the `least priority`.
+</font>
+
+## Data Exploration of all the tables in the database
+
+### Data Exploartion of bugs_mod Table from the Database
+
+```{r bugs mod connection}
+bugs_mod_df <- tbl(con, "bugs_mod")
+# Converting `bugs_mod_df to `dataframe`
+bugs_mod_df <- as.data.frame(bugs_mod_df)
+#for quick view of the datatypes and the structure of data
+skim(bugs_mod_df)
```

Comment:
This is okay for a first look, but it would be better to summarise the differences between `bugs_mod_df` and `bugs`: most variables are unchanged; there is the additional rownames variable, and some variables are a different data type (some now all NA). Is there any additional information here, or can this table be ignored?

Reply:
`rownames` variable is the serial number and all data of `bugs_mod` are similar to the `bugs` data table. So, this table can be ignored.

Reply:
I understand this, but the table is part of the database dump, so it would be helpful for you as the analyst to briefly summarise the differences in the the report, rather than just showing the skimr summary and leaving people to work it out for themselves. Something along the lines of "`bug_mod_df` is a modified version of the `bugs` data table. It has an additional `rownames` variable, that is simply the number of the table row. Otherwise the variables are the same, but [x, y, z] are recorded as [dates] rather than [character] and [p,q] are recorded as [character] rather than [numeric]. Since these variables have been analysed above, it is not necessary to consider this table further". (It does raise the question of whether it would have been easier to use this modified table in the first place though, perhaps the data types here are what we want, so you could have avoided doing the data type conversion in the earlier analysis? It would be good to know to simplify future analyses, which is why it is helpful to summarise the differences).